packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/platform_interface/google_maps_flutter_platform.dart

```diff
@@ -443,4 +443,12 @@ abstract class GoogleMapsFlutterPlatform extends PlatformInterface {
       mapOptions: jsonForMapConfiguration(mapConfiguration),
     );
   }
+
+  /// Populates [GoogleMapsFlutterInspectorPlatform.instance] to allow
+  /// inspecting the platform map state.
+  @visibleForTesting
+  void enableDebugInspection() {
```

Comment:
@ditman Does this structure address https://github.com/flutter/flutter/issues/80688#issuecomment-853288237?

My thought was that by having the population controlled from here, you should be able to do whatever bridging is necessary between the private data you mentioned and the web implementation of the inspector. E.g., its constructor could take a reference to the "very private" data you mentioned, or if that's too much access, maybe to a provider object that can access pieces of it via lambas?

Reply:
I think this should be enough. The end goal is to "leak" this `_googleMap` object to whatever `extends GoogleMapsInspectorPlatform` we created for the web:

https://github.com/flutter/plugins/blob/main/packages/google_maps_flutter/google_maps_flutter_web/lib/src/google_maps_controller.dart#L86

Which is wrapped by a Controller object that is indexed by mapId here:

https://github.com/flutter/plugins/blob/main/packages/google_maps_flutter/google_maps_flutter_web/lib/src/google_maps_flutter_web.dart#L17

(Maybe this method can just enable a boolean that lets us user a _googleMap getter?)